### PR TITLE
Version v0.3.0

### PR DESCRIPTION
**Pull request by prescottprue**
* Use `react-redux-firebase` (v1.4.0) in place of `redux-devshare` for data binding
* Implemented auth decorators for redirecting
* Updated `redux-devshare` to `v0.3.0`
* Anonymous projects fixed
* files structure is now its own component

### CODE REVIEW

File: src/routes/Projects/routes/Project/containers/TreeView/TreeView.js [hunk 1]
```diff
@@ -38,7 +30,7 @@ export default class TreeView extends Component {
 
   static propTypes = {
     project: PropTypes.object.isRequired,
-    files: PropTypes.array,
+    files: PropTypes.object,
```
Comment (sbolel) [inline src/routes/Projects/routes/Project/containers/TreeView/TreeView.js]: Yep, I also made this change in #57 haha

File: src/routes/Projects/routes/Project/containers/TreeView/TreeView.js [hunk 2]
```diff
@@ -81,10 +74,10 @@ export default class TreeView extends Component {
   }
 
   getStructure = () => {
-    if (!this.props.files || !this.props.files.length) {
+    if (!this.props.files || !size(this.props.files)) {
       return null
     }
-    return this.props.files.map((entry, i) => {
+    return map(this.props.files, (entry, i) => {
```
Comment (sbolel) [inline src/routes/Projects/routes/Project/containers/TreeView/TreeView.js]: Why the preference for lodash over native?

Comment (prescottprue) [inline src/routes/Projects/routes/Project/containers/TreeView/TreeView.js]: Lodash `map` works for objects. Then the `i` becomes the key of the item within Firebase

Comment (sbolel) [inline src/routes/Projects/routes/Project/containers/TreeView/TreeView.js]: ah. woudja lookathat!

**Comment by prescottprue** [comment]
This is still not totally working, so I am going to hold it up.

**Pull request merged**